### PR TITLE
ENH Replace gpuci_conda_retry with gpuci_mamba_retry

### PR DESCRIPTION
**Pull request by dillon-cullinan**
`mamba` was recently added to gpuCI build environment, testing usage and solvability with this PR which should speed up build times.

### CODE REVIEW

**Comment by dillon-cullinan** [comment]
rerun tests

**Comment by robertmaynard** [comment]
@dillon-cullinan What is this waiting on anything for us to merge?

**Comment by dillon-cullinan** [comment]
> @dillon-cullinan What is this waiting on anything for us to merge?

I created this PR in bulk to other repos as well. Only two are failing, I want to make sure mamba works everywhere before merging these. Once they're all passing I'll get them reviewed by other OPS members.

**Review by ajschmidt8 — APPROVED** [review]

**Comment by dillon-cullinan** [comment]
@gpucibot merge

**Pull request merged**
